src/features/metadata-based-view/MetadataQueryAPIHelper.js

```diff
@@ -75,26 +117,27 @@ export default class MetadataQueryAPIHelper {
             return {};
         }
 
-        const fields = Object.keys(instance)
-            .filter(key => !key.startsWith('$'))
-            .map(key => {
-                const templateField = find(templateFields, ['key', key]);
-                const type = getProp(templateField, 'type'); // get data type
-                const displayName = getProp(templateField, 'displayName'); // get displayName
-                const field: MetadataQueryInstanceTypeField = {
-                    key,
-                    value: instance[key],
-                    type,
-                    displayName,
-                };
-
-                if (includes(SELECT_TYPES, type)) {
-                    // get "options" for enums or multiselects
-                    field.options = getProp(templateField, 'options');
-                }
-
-                return field;
-            });
+        const queryFields = this.getMetadataQueryFields();
+
+        const fields = queryFields.map((queryField: string) => {
```

Comment:
Iterate over the fields mentioned in the query instead of fields returned in the response. The `metadata_queries` API will return *only* the existing fields.